juniper_official/Subscriber-Service/address-pool-utilization.rule

```diff
@@ -0,0 +1,292 @@
+                description "address pool utilization minimum threshold value";
+                type int;
+            }
+        }
+        rule linked-address-pool {
+            synopsis "linked address pool utilization analyzer";
+            description "Monitors linked address pool usage percentage and notifies anomalies";
+            rule-frequency 90s;
+            function mean-pool-utilization {
+                path mean-pool-utilization.py;
+                method mean_pool_utilization;
+                argument list_pool_vector {
+                    mandatory;
+                }
+            }
+            field linked-pool-head {
+                constant {
+                    value "{{filter-linked-pool-head}}";
+                }
+                type string;
+                description "linked-pool-head name field";
+            }
+            field mean-pool-utilization {
+                formula {
+                    user-defined-function {
+                        function-name mean-pool-utilization;
+                        argument list_pool_vector "@list-pool-utilization";
+                    }
+                }
+                type integer;
+                description "average utilizaiton of linked address pool";
+            }
+            field mean-pool-utilization-ml {
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 1d;
+                        pattern-periodicity 1h;
+                        field-name "$mean-pool-utilization";
+                    }
+                }
+                type integer;
+                description "Value calculated based on 3sigma algorithm";
+            }
+            field pool-maximum-utilization {
+                constant {
+                    value "{{pool-max-utilization}}";
+                }
+                description "linked address pool utilization maximum threshold";
+            }
+            field pool-minimum-util {
+                constant {
+                    value "{{pool-min-utilization}}";
+                }
+                description "linked address pool utilization minimum threshold";
+            }
+            trigger mean-pool-utilization {
+                synopsis "linked address pool utilization kpi";
+                description "Sets health based on linked address pool utilization";
+                frequency 120s;
+                term is-mean-abnormal {
+                    when {
+                        greater-than "$mean-pool-utilization" "$pool-maximum-utilization";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Linked pool average utitilization($mean-pool-utilization) is abnormal.";
+                        }
+                    }
+                }
+                term mean-abnormal-ls {
```

Comment:
Can you please re-phrase this term name to something more meaningful?
If there is a "when", the name should be a verb-phrase-question.

Reply:
I made it like "is-mean-abnormal-lt". Not able to make more meaningful because of number of character limit 24.